examples/bridge-app/linux/main.cpp

```diff
@@ -452,16 +413,33 @@ void HandleDevicePowerSourceStatusChanged(DevicePowerSource * dev, DevicePowerSo
     if (itemChangedMask & DevicePowerSource::kChanged_BatLevel)
     {
         uint8_t batChargeLevel = dev->GetBatChargeLevel();
+        DeviceLayer::StackLock lock;
         MatterReportingAttributeChangeCallback(dev->GetEndpointId(), PowerSource::Id, PowerSource::Attributes::BatChargeLevel::Id,
                                                ZCL_INT8U_ATTRIBUTE_TYPE, &batChargeLevel);
     }
 
     if (itemChangedMask & DevicePowerSource::kChanged_Description)
     {
+        DeviceLayer::StackLock lock;
         MatterReportingAttributeChangeCallback(dev->GetEndpointId(), PowerSource::Id, PowerSource::Attributes::Description::Id);
     }
 }
 
+void HandleDeviceTempSensorStatusChanged(DeviceTempSensor * dev, DeviceTempSensor::Changed_t itemChangedMask)
+{
+    if (itemChangedMask &
+        (DeviceTempSensor::kChanged_Reachable | DeviceTempSensor::kChanged_Name | DeviceTempSensor::kChanged_Location))
+    {
+        HandleDeviceStatusChanged(static_cast<Device *>(dev), (Device::Changed_t) itemChangedMask);
+    }
+    if (itemChangedMask & DeviceTempSensor::kChanged_MeasurementValue)
+    {
+        DeviceLayer::StackLock lock;
+        MatterReportingAttributeChangeCallback(dev->GetEndpointId(), TemperatureMeasurement::Id,
```

Comment:
Please see how the callbacks are done on tip.

Reply:
Done, Thanks for the changes you made for this.